private/buf/bufgen/generator.go

```diff
@@ -209,34 +208,12 @@ func (g *generator) execPlugins(
 	for remote, indexedPluginConfigs := range remotePluginConfigTable {
 		remote := remote
 		indexedPluginConfigs := indexedPluginConfigs
-		v1Args := make([]*remotePluginExecArgs, 0, len(indexedPluginConfigs))
 		v2Args := make([]*remotePluginExecArgs, 0, len(indexedPluginConfigs))
 		for _, param := range indexedPluginConfigs {
-			if param.PluginConfig.Plugin == "" {
-				v1Args = append(v1Args, param)
-			} else {
-				v2Args = append(v2Args, param)
+			if param.PluginConfig.Remote != "" {
+				return nil, fmt.Errorf("invalid plugin reference: %s", param.PluginConfig.Remote)
```

Comment:
We don't bother a roundtrip to the server, and instead fail if we detect any legacy plugin that has remote.

Suggestions are welcome for a nicer error message.

Given the work @oliversun09 is doing, I think we should avoid a large refactor within this code path.